src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -0,0 +1,561 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+from __future__ import print_function, unicode_literals, division, absolute_import
+
+import sys
+import collections
+import json
+import os
+import re
+import csv
+import dxpy
+import pandas as pd
+from ..utils.printing import (fill)
+from ..bindings import DXRecord
+from ..bindings.dxdataobject_functions import is_dxlink
+from ..bindings.dxfile import DXFile
+from ..utils.resolver import resolve_existing_path, ResolutionError
+from ..utils.file_handle import as_handle
+from ..exceptions import err_exit, PermissionDenied, InvalidInput, InvalidState
+
+database_unique_name_regex = re.compile('^database_\w{24}__\w+$')
+database_id_regex = re.compile('^database-\\w{24}$')
+
+def extract_dataset(args):
+    """
+       (Preview feature) Retrieves the data, or generates SQL to retrieve the data, from a dataset or cohort for a set of entity.fields. 
+    """
+    if not args.dump_dataset_dictionary and args.fields is None:
+        raise err_exit(fill('Must provide at least one of the following options: --fields or --dump-dataset-dictionary'))
+
+    if len(args.delim) == 1:
+        delimiter = str(args.delim)
+        if delimiter == ",":
+            out_extension = ".csv"
+        elif delimiter == "\t":
+            out_extension = ".tsv"
+        else:
+            out_extension = ".txt"
+    else:
+        raise err_exit(fill('Invalid delimiter specified'))
+    
+    project, path, entity_result = resolve_existing_path(args.path)
+
+    try:
+        resp = dxpy.DXHTTPRequest('/' + entity_result['id'] + '/visualize',
+                                        {"project": project, "cohortBrowser": False} )
+    except PermissionDenied:
+        print("Insufficient permissions")
+        sys.exit(1)
```

Comment:
How about calling `err_exit`, e.g. like [here](https://github.com/dnanexus/dx-toolkit/pull/786/files#diff-9fa6c7c89cfc29f5fe6314b6c36a1dfab692ac8a3378d7e1d90e244ed233d40bR77)

Reply:
`err_exit` raises an exception which will again be caught by the next except statement. We are trying to keep error messages simple and not print stacktrace wherever possible. Updated the except statement below this from ResolutionError to err_exit.

Reply:
`err_exit()` can be provided expected exceptions and then will only print the stacktrace if `dxpy._DEBUG > 1`
e.g. https://github.com/dnanexus/dx-toolkit/blob/d321b315d4c03c5aab9125b083544ed3e35cd051/src/python/dxpy/scripts/dx.py#L1491-L1492

Reply:
Updated.